feat: Implement gamification system

lesbiannix opened the pull request:
This commit introduces a comprehensive gamification system to the PsychonautWiki Journal application.

The new system includes:
- XP-based leveling system
- Achievement system
- Streak tracking
- Safety score calculation
- Knowledge quest system
- Weekly safety challenges
- AI-powered personalized insights

The implementation includes data models, services, UI components, and integration into the existing application.

google-labs-jules[bot] committed:
feat: Implement gamification system

This commit introduces a comprehensive gamification system to the PsychonautWiki Journal application.

The new system includes:
- XP-based leveling system
- Achievement system
- Streak tracking
- Safety score calculation
- Knowledge quest system
- Weekly safety challenges
- AI-powered personalized insights

The implementation includes data models, services, UI components, and integration into the existing application.

harm-reduction-a11y commented:
Are you serious? Are you **SERIOUS**?? "Imbecilic" doesn't even begin to describe the depth to which your stupidity reaches. Please detail why you think gamifying and adding a generative AI to a *harm reduction* app is anything other than a HORRIBLE idea. This is how people die. Please leave the profession of software engineering to those with grey and white matter between their ears, because it's clear to me that this is something you lack.

I refuse to believe this is anything other than a sick joke. You even vibe coded it together. Why, oh god why, do you think it's a good idea to [reward logging streaks](https://github.com/isaakhanimann/psychonautwiki-journal-android/blob/d1bc65ba8d8f9714322da2a144754b8d2ccc274f/app/src/main/java/com/isaakhanimann/journal/data/gamification/WeeklyChallengeServiceImpl.kt#L16)? Seriously?? Are you for real??? People don't only use this shit for medications you know. I'd tell you to use your brain, but...

Did you even read the code before submitting? Did you even take a single fucking second to think that it might be a bad idea to potentially feed people faulty information regarding harm reduction via a generative AI? Are you okay? *Are you okay???*

I know normies more mindful than you. Please consider closing this pull request, as well as deleting your GitHub account.

Thanks! ✨

lesbiannix committed:
Attempt to fix Android compilation issues (#1)

This change represents an attempt to resolve the Android compilation issues. The following steps were taken:
- Installed the Android SDK and command-line tools.
- Accepted all SDK licenses.
- Installed missing build-tools and platform components.
- Corrected SDK directory permissions.

Despite these efforts, the `./gradlew build` command continues to fail or hang without providing a useful error log. However, the `./gradlew test` command succeeds.

Submitting the current state to allow the user to take over the debugging process on their local machine.

Co-authored-by: google-labs-jules[bot] <161369871+google-labs-jules[bot]@users.noreply.github.com>

lesbiannix committed:
Delete build.log

isaakhanimann commented:
Thank you for putting in the work. Unfortunately I can't include it in the private repo where I'm continuing to develop the app. Sorry.
